experiments/experiments-impl/src/main/java/com/duckduckgo/experiments/impl/VariantManagerImpl.kt

```diff
@@ -106,12 +106,20 @@ class VariantManagerImpl @Inject constructor(
         if (entity.key == "sc" || entity.key == "se") {
             return { isSerpRegionToggleCountry() }
         }
-        if (entity.localeFilter.isEmpty()) {
-            return { noFilter() }
+
```

Comment:
All this logic needs testing. We cannot write critical code without creating tests for it.

Reply:
You are right, the problem is that filters are set in a private function within the class. I suppose I can create another layer for experiment filters and test it properly.

Reply:
I think creating a new class to manage filters is a better approach that scales much better. We'll be adding more, like country or language, so I think it's totally worth creating the new class now and not overcomplicating the Test class